### PR TITLE
`model.config`: Align with breaking changes in the Hub

### PR DESCRIPTION
**Pull request by julien-c**
following (internal) https://github.com/huggingface/moon-landing/pull/8978

### CODE REVIEW

**Review by SBrandeis — APPROVED** [review]

File: packages/tasks/src/model-data.ts [hunk 1]
```diff
@@ -40,21 +40,29 @@ export interface ModelData {
...
+		adapter_transformers?: {
+			model_name?: string;
+			model_class?: string;
+		};
+		diffusers?: {
+			_class_name?: string;
+		};
 		sklearn?: {
-			filename?: string;
+			model?: {
```
Comment (osanseviero) [inline packages/tasks/src/model-data.ts:54]: Why this change? Isn't it `filename` in the Hub?

Comment (julien-c) [reply packages/tasks/src/model-data.ts:54]: nope not after https://github.com/huggingface/moon-landing/pull/8978 where we enforce that we always use the prop names from the actual JSON files (will make things simpler in the long run)

File: packages/tasks/src/model-data.ts [hunk 2]
```diff
@@ -40,21 +40,29 @@ export interface ModelData {
...
+				file?: string;
+			};
 			model_format?: string;
 		};
 		speechbrain?: {
-			interface?: string;
+			speechbrain_interface?: string;
```
Comment (osanseviero) [inline packages/tasks/src/model-data.ts:60]: Btw I just realized the `speechbrain` API needs some additional fields https://github.com/huggingface/api-inference-community/blob/main/docker_images/speechbrain/app/common.py#L39

Comment (julien-c) [reply packages/tasks/src/model-data.ts:60]: `vocoder_model_id` is also in there, i'll add it to types

Comment (julien-c) [reply packages/tasks/src/model-data.ts:60]: see 14ffcb7b740378e16a3a2bdf93325235528f70d7

Comment (osanseviero) [reply packages/tasks/src/model-data.ts:60]: Ah nice, nothing to worry then!

File: packages/tasks/src/model-data.ts (hunk 1 above)
Comment (osanseviero) [reply packages/tasks/src/model-data.ts:54]: Related PR https://github.com/huggingface/moon-landing/pull/9104
